src/language/rule/rule.ts

```diff
@@ -265,3 +278,37 @@ export class RuleFailure {
         return new RuleFailurePosition(position, lineAndCharacter);
     }
 }
+
+export class WalkContext<T> {
+    private limit: number;
+    private failures: RuleFailure[];
+
+    constructor(public readonly sourceFile: ts.SourceFile, public readonly ruleName: string, public readonly options: T) {
+        this.failures = [];
+        this.limit = sourceFile.getFullWidth();
+    }
+
+    public getFailures() {
```

Comment:
The failures array could just be declared as `public readonly failures: RuleFailure[] = []` and let the AbstractWalker implement the `getFailures` method.